src/engine/renderer/InternalImage.cpp

```diff
@@ -0,0 +1,129 @@
+/*
+===========================================================================
+
+Daemon BSD Source Code
+Copyright (c) 2013-2016 Daemon Developers
+All rights reserved.
+
+This file is part of the Daemon BSD Source Code (Daemon Source Code).
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of the Daemon developers nor the
+      names of its contributors may be used to endorse or promote products
+      derived from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL DAEMON DEVELOPERS BE LIABLE FOR ANY
+DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+===========================================================================
+*/
+// InternalImage.cpp
+#include "tr_local.h"
+
+// Comments may include short quotes from other authors.
+
+int R_GetImageCustomScalingStep( const image_t *image, imageParams_t *imageParams )
+{
+	int scalingStep = 0;
+
+	// Perform optional picmip operation.
+	if ( !( image->bits & IF_NOPICMIP ) )
+	{
+		int imageMinDimension = std::max( -1, r_imageMinDimension->integer );
+		int imageMaxDimension = std::max( 0, r_imageMaxDimension->integer );
+		int shaderMinDimension = 0;
+		int shaderMaxDimension = 0;
+
+		if ( imageParams != nullptr )
```

Comment:
Done, except for functions calling `R_LoadImage` (`R_FindImageFile` and `R_FindCubeImage`) because it would have required to also fix all image loaders (`LoadWEBP`, `LoadPNG`, `LoadTGA`, `LoadJPG`, `LoadDDS`, `LoadCRN`, `LoadKTX`) and many of the function they call, and this PR would then edit a dozen of files…